util/loadutil/disk.go

```diff
@@ -0,0 +1,285 @@
+}
+
+func GetMatchParation(path string) (*disk.PartitionStat, error) {
+	partitons, err := disk.Partitions(true)
+	if err != nil {
+		return nil, err
+	}
+	maxMatch := 0
+	matchParation := disk.PartitionStat{}
+	for _, partition := range partitons {
+		match := getMatchCount(path, partition.Mountpoint)
+		if match == len(partition.Mountpoint) && match > maxMatch {
+			matchParation = partition
+		}
+	}
+	return &matchParation, nil
+}
+
+var (
+	ErrInvaliPartition      = errors.New("invalid partiton")
+	ErrFailedToGetIoCounter = errors.New("failed to get io counter")
+)
+
+func getDeviceNameFromPartition(partition *disk.PartitionStat) (string, error) {
+	var name string
+	if n, _ := fmt.Sscanf(partition.Device, "/dev/%s", &name); n != 1 {
+		return "", ErrInvaliPartition
+	}
+	return name, nil
+}
+
+func GetIoCounter(partition *disk.PartitionStat) (*disk.IOCountersStat, error) {
+	name, err := getDeviceNameFromPartition(partition)
+	if err != nil {
+		return nil, err
+	}
+	counters, err := disk.IOCounters(name)
+	if err != nil {
+		return nil, err
+	}
+	counter, exist := counters[name]
+	if !exist {
+		return nil, ErrFailedToGetIoCounter
+	}
+	return &counter, nil
+}
+
+type DiskIoSampleItem struct {
+	time      time.Time
+	ioCounter *disk.IOCountersStat
+}
+
+func getDiskIoSampleItem(partition *disk.PartitionStat) (*DiskIoSampleItem, error) {
+	ioCounter, err := GetIoCounter(partition)
+	if err != nil {
+		return nil, err
+	}
+	time := time.Now()
+	return &DiskIoSampleItem{
+		time:      time,
+		ioCounter: ioCounter,
+	}, nil
+}
+
+func getReadFlow(first *DiskIoSampleItem, second *DiskIoSampleItem) uint64 {
+	t := second.time.Sub(first.time)
+	ms := uint64(t.Milliseconds())
+	bytes := second.ioCounter.ReadBytes - first.ioCounter.ReadBytes
+	return bytes * 1000 / ms
+}
+
+func getWriteFlow(first *DiskIoSampleItem, second *DiskIoSampleItem) uint64 {
+	t := second.time.Sub(first.time)
+	ms := uint64(t.Milliseconds())
+	bytes := second.ioCounter.WriteBytes - first.ioCounter.WriteBytes
+	return bytes * 1000 / ms
+}
+
+func getIoCount(first *DiskIoSampleItem, second *DiskIoSampleItem) uint64 {
+	count := second.ioCounter.ReadCount - first.ioCounter.ReadCount + second.ioCounter.WriteCount - first.ioCounter.WriteCount
+	return count
+}
+
+func getTotalReadWaitTime(first *DiskIoSampleItem, second *DiskIoSampleItem) uint64 {
+	count := second.ioCounter.ReadTime - first.ioCounter.ReadTime
+	return count
+}
+
+func getTotalWriteWaitTime(first *DiskIoSampleItem, second *DiskIoSampleItem) uint64 {
+	count := second.ioCounter.WriteTime - first.ioCounter.WriteTime
+	return count
+}
+
+func getIoTotalWaitTime(first *DiskIoSampleItem, second *DiskIoSampleItem) uint64 {
+	count := second.ioCounter.IoTime - first.ioCounter.IoTime
+	return count
+}
+
+func getIoTotalWeightedWaitTime(first *DiskIoSampleItem, second *DiskIoSampleItem) uint64 {
+	count := second.ioCounter.WeightedIO - first.ioCounter.WeightedIO
+	return count
+}
+
+type DiskIoSample struct {
```

Comment:
remove DiskIoSample, you can check MetricIOBytes

Reply:
as same as prev